### ISSUE TITLE
Amplify UI React issue with invalid TOTP code

### ISSUE THREAD

**Issue by gordonbrown** [aws-amplify/amplify-ui#498]
# JavaScript Framework
React

# Amplify APIs
 - @aws-amplify/ui-components: ^1.7.9
 - @aws-amplify/ui-react: ^1.2.16
 - aws-amplify: ^4.2.10

 # Environment Information
 ## System
 - OS: Windows 10 10.0.19043
 - CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
 - Memory: 10.71 GB / 31.72 GB

 # Browser
  - Chrome: 93.0.4577.63 (Official Build) (64-bit)

# Cognito User Pool Setup
 - MFA: required
 - Second factors enabled: Time-based One-time Password only

 # Code Setup
 Adapted from the AWS documentation on recommended usage of authenticator component for React
https://docs.amplify.aws/ui/auth/authenticator/q/framework/react/#recommended-usage 

```
import React from "react";
import Amplify from "aws-amplify";
import { AmplifyAuthenticator, AmplifySignOut } from "@aws-amplify/ui-react";
import { AuthState, onAuthUIStateChange } from "@aws-amplify/ui-components";
import config from "./config";

Amplify.configure({
	Auth: {
		mandatorySignIn: true,
		region: config.cognito.REGION,
		userPoolId: config.cognito.USER_POOL_ID,
		identityPoolId: config.cognito.IDENTITY_POOL_ID,
		userPoolWebClientId: config.cognito.APP_CLIENT_ID,
		authenticationFlowType: config.cognito.USER_PASSWORD_AUTH,
	},
});
const App = () => {
	const [authState, setAuthState] = React.useState();
	const [user, setUser] = React.useState();

	React.useEffect(() => {
		return onAuthUIStateChange((nextAuthState, authData) => {
			setAuthState(nextAuthState);
			setUser(authData);
		});
	}, []);

	return authState === AuthState.SignedIn && user ? (
		<div className="App">
			<div>Hello, {user.username}</div>
			<AmplifySignOut />
		</div>
	) : (
		<AmplifyAuthenticator />
	);
};

export default App;
```
# Description

## Summary
When a user enters an invalid TOTP verification code, no error message is displayed to the user.

## Steps
 - A user has an account in Cognito but MFA has not been set up
 - User is presented with the login screen (1)
 - User enters valid username and password and clicks Sign in (2)
 - Upon entering an invalid security code and clicking Verify Security Token, no toast error message is displayed to the user. However, the browser console does show the correct details of the error:
  
`[ERROR] 39:18.363 TOTP {code: 'EnableSoftwareTokenMFAException', name: 'EnableSoftwareTokenMFAException', message: 'Code mismatch and fail enable Software Token MFA'}`

## Attachments
1. <img width="496" alt="Screenshot 2021-10-12 at 07 17 53" src="https://user-images.githubusercontent.com/1052606/136903302-d0b0d809-729f-42b9-9c6d-a273dbfe56cd.png">

2. <img width="484" alt="Screenshot 2021-10-12 at 07 19 14" src="https://user-images.githubusercontent.com/1052606/136903380-4664e4d9-e3f5-4447-bc6f-2a42258b48c9.png">

**Comment by ericclemmons** [comment]
> Upon entering an invalid security code and clicking Verify Security Token, no toast error message is displayed to the user. However, the browser console does show the correct details of the error:


Closing as we resolved with with our `@next` release 🙏

**Comment by StevenHee** [comment]
Thanks for the fix in the @next release and apologies for the delay in my response. I have been able to reproduce the error message locally using the code example you provided. However, it only partially solves our problem. We have implemented the individual UI components in our application which don’t appear to be available in the @next release and are unable to test the changes in situ. Will the individual components be available when the @next is released to production? Also, we will want to be able display a modified error message to users so wanted to check if this error is now available via the amplify Hub eventing system?

**Comment by ericclemmons** [comment]
Which components are you customizing?

(Reopening)

**Comment by StevenHee** [comment]
We're trying to use the UI components out-of-the-box rather than rolling our own, so we're currently importing the following from `@aws-amplify/ui-react`

- AmplifyAuthContainer,
- AmplifyAuthenticator,
- AmplifySignIn,
- AmplifyTotpSetup,
- AmplifyConfirmSignIn,
- AmplifyForgotPassword,
- AmplifyRequireNewPassword

**Comment by StevenHee** [comment]
Apart from styling changes, the only customisation is the text displayed in the header of each component, which the individual UI components allows us to modify via the `headerText` property.

**Comment by ericclemmons** [comment]
Oh! Do these docs cover that use-case then?

> https://ui.docs.amplify.aws/ui/components/authenticator?platform=react#labels--text

**Comment by StevenHee** [comment]
Thanks Eric. Yes, that will probably resolve the header text issue. However, we don't wrap our entire application with authentication; presenting intro text as well as customizing the navigation in header and footer depending whether the user is logged in, which the UI components allow us to do. You can view our application landing page at https://trainee.tis-selfservice.nhs.uk

**Comment by ericclemmons** [comment]
@StevenHee You've been a huge help!  I think I should update most of the examples to use `<Authenticator>{props => ...}</Authenticator>`, since the "full-screen modal" is just one way of using it (#576).

I also opened up https://github.com/aws-amplify/amplify-ui/issues/637 that should give you full control to customize around the form.

Let's discuss more in #637 how that can meet your customization needs. (Your Authenticator implementation code help me understand better, too) 🙏 

In the meantime, since the `@next` version does resolve this TOTP bug, I'm going to close this issue so the team is aware this bug has been addressed.
